app/api/resources/metadata.py

```diff
@@ -104,16 +104,37 @@ def example_objects(request: Request, genome_id: str):
 
 @router.get("/genome/{genome_uuid}/details", name="genome_details")
 async def get_genome_details(request: Request, genome_uuid: str):
-    genome_details_dict = MessageToDict(grpc_client.get_genome_details(genome_uuid))
-    genome_details = GenomeDetails(**genome_details_dict)
-    return responses.JSONResponse(genome_details.dict())
+    response_data = None
+    try:
+        genome_details_dict = MessageToDict(grpc_client.get_genome_details(genome_uuid))
+        if genome_details_dict:
+            genome_details = GenomeDetails(**genome_details_dict)
+            response_data = responses.JSONResponse(genome_details.dict())
+        else:
+            not_found_response = {"message": "Could not find details for {}".format(genome_uuid)}
+            response_data = responses.JSONResponse(not_found_response, status_code=404)
+    except Exception as ex:
+        logger.log("DEBUG", ex)
+    return response_data
 
 @router.get("/genome/{slug}/explain", name="genome_explain")
 async def explain_genome(request: Request, slug: str):
-    genome_uuid = grpc_client.get_genome_uuid_from_slug(slug)
+    uuid_from_tag = grpc_client.get_genome_uuid_from_tag(slug)
     response_dict = {}
-    if genome_uuid:
-        genome_details_dict = MessageToDict(grpc_client.get_genome_details(genome_uuid))
-        genome_details = GenomeDetails(**genome_details_dict)
-        response_dict = genome_details.model_dump(include={"genome_id":True, "genome_tag":True, "scientific_name":True, "common_name":True, "is_reference" : True, "assembly": {"name", "accession_id"}, "type": True})
-    return responses.JSONResponse(response_dict)
+    response_data = None
+    if uuid_from_tag:
+        genome_detail_for_genome = uuid_from_tag
```

Comment:
`genome_detail_for_genome` is just a `genome_uuid`, isn't it? A string that we hope (but not perfectly sure; but that's all right) is a genome uuid.

Would it be easier if line 122 said `genome_uuid = grpc_client.get_genome_uuid_from_slug(slug) or slug`?